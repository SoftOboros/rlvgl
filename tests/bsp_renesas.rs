//! Round-trip test for the Renesas YAML adapter.
#![cfg(all(feature = "creator", feature = "regression"))]

#[path = "../src/bin/creator/bsp/ir.rs"]
mod ir;
#[path = "../src/bin/creator/bsp/renesas.rs"]
mod renesas;

use minijinja::{Environment, context};

#[test]
fn renesas_roundtrip_snapshot() {
    let yaml = include_str!("fixtures/renesas.yaml");
    let spec = renesas::yaml_to_ir(yaml).unwrap();
    insta::assert_yaml_snapshot!("ir", &spec);

    let mut env = Environment::new();
    env.add_template(
        "gen",
        include_str!("../src/bin/creator/bsp/templates/simple.rs.jinja"),
    )
    .unwrap();
    let tmpl = env.get_template("gen").unwrap();
    let rendered = tmpl.render(context! { spec => &spec }).unwrap();
    insta::assert_snapshot!("generated", rendered);
}
