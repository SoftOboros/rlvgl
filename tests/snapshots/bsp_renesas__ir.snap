---
source: tests/bsp_renesas.rs
expression: "&spec"
---
mcu: RA6M4
package: LQFP64
clocks:
  pll: {}
  kernels: {}
pinctrl:
  - pin: P300
    func: SPI0_SCK
    af: 0
  - pin: P301
    func: SPI0_MOSI
    af: 0
peripherals:
  spi0:
    class: spi
    signals:
      sck: P300
      mosi: P301
